src/openrct2-ui/windows/GameBottomToolbar.cpp

```diff
@@ -69,687 +69,672 @@ static Widget window_game_bottom_toolbar_widgets[] =
+                   ? COLOUR_WHITE
+                   : NOT_TRANSLUCENT(colours[0]));
+        StringId stringId = DateFormatStringFormatIds[gConfigGeneral.DateFormat];
+        auto ft = Formatter();
+        ft.Add<StringId>(DateDayNames[day]);
+        ft.Add<int16_t>(month);
+        ft.Add<int16_t>(year);
+        DrawTextBasic(dpi, screenCoords, stringId, ft, { colour, TextAlignment::CENTRE });
+
+        // Figure out how much line height we have to work with.
+        uint32_t line_height = FontGetLineHeight(FontStyle::Medium);
 
-                WindowBase* mainWindow = WindowGetMain();
-                if (mainWindow != nullptr)
-                    WindowScrollToLocation(*mainWindow, subjectLoc.value());
+        // Temperature
+        screenCoords = { windowPos.x + window_game_bottom_toolbar_widgets[WIDX_RIGHT_OUTSET].left + 15,
+                         static_cast<int32_t>(screenCoords.y + line_height + 1) };
+
+        int32_t temperature = gClimateCurrent.Temperature;
+        StringId format = STR_CELSIUS_VALUE;
+        if (gConfigGeneral.TemperatureFormat == TemperatureUnit::Fahrenheit)
+        {
+            temperature = ClimateCelsiusToFahrenheit(temperature);
+            format = STR_FAHRENHEIT_VALUE;
+        }
+        ft = Formatter();
+        ft.Add<int16_t>(temperature);
+        DrawTextBasic(dpi, screenCoords + ScreenCoordsXY{ 0, 6 }, format, ft);
+        screenCoords.x += 30;
+
+        // Current weather
+        auto currentWeatherSpriteId = ClimateGetWeatherSpriteId(gClimateCurrent);
+        GfxDrawSprite(&dpi, ImageId(currentWeatherSpriteId), screenCoords);
+
+        // Next weather
+        auto nextWeatherSpriteId = ClimateGetWeatherSpriteId(gClimateNext);
+        if (currentWeatherSpriteId != nextWeatherSpriteId)
+        {
+            if (gClimateUpdateTimer < 960)
+            {
+                GfxDrawSprite(&dpi, ImageId(SPR_NEXT_WEATHER), screenCoords + ScreenCoordsXY{ 27, 5 });
+                GfxDrawSprite(&dpi, ImageId(nextWeatherSpriteId), screenCoords + ScreenCoordsXY{ 40, 0 });
             }
-            break;
-        case WIDX_RIGHT_OUTSET:
-        case WIDX_DATE:
-            ContextOpenWindow(WindowClass::RecentNews);
-            break;
+        }
     }
-}
 
-static OpenRCT2String WindowGameBottomToolbarTooltip(WindowBase* w, const WidgetIndex widgetIndex, const StringId fallback)
-{
-    auto ft = Formatter();
-
-    switch (widgetIndex)
+    /**
+     *
+     *  rct2: 0x0066BFA5
+     */
+    void DrawNewsItem(DrawPixelInfo &dpi)
     {
-        case WIDX_MONEY:
-            ft.Add<money64>(gCurrentProfit);
-            ft.Add<money64>(gParkValue);
-            break;
-        case WIDX_PARK_RATING:
-            ft.Add<int16_t>(gParkRating);
-            break;
-    }
-    return { fallback, ft };
-}
+        int32_t itemWidth;
+        News::Item* newsItem;
+        Widget* middleOutsetWidget;
```

Comment:
Declare and define those as needed (this is legacy from C times). An example below.